### PR TITLE
(DOCSP-25673): Link credentials Kotlin

### PR DESCRIPTION
**Pull request by cbullinger**
### Jira

- https://jira.mongodb.org/browse/DOCSP-25673

### Staged Changes (Requires MongoDB Corp SSO)

- [Link User Identities](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-25673-link-credentials-kotlin/sdk/kotlin/app-services/link-credentials/)

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/sdk/kotlin/app-services/link-credentials.txt [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Realm provides :ref:`many authentication providers <authentication-providers>` 
+to log users into your app. Each provider creates a unique user identity. 
+Realm lets you merge multiple credentials into one user identity.
```
Comment (cmelchior) [inline source/sdk/kotlin/app-services/link-credentials.txt:15]: I think there needs to be some description of caveats here, i.e. you cannot link credentials if it was already used to login (and thus fully create the user). So this would not work:

```
val anonUser = app.login(Credentials.anonymous()) // logs in with an anonymous user
app.emailPasswordAuth.registerUser(email, password)
val credentials = Credentials.emailPassword(email, password)
val emailUser = app.login(credentials)
anonUser.linkCredentials(credentials); // Will fail
```

I believe it is the same for other auth providers as well. If you once logged in with them (and thus fully created a user). You can no longer merge them.

Comment (cbullinger) [reply source/sdk/kotlin/app-services/link-credentials.txt]: Good catch. I've updated to specify that you can't link credentials if they've already been used to login a user and that you can't link multiple email/pwd credentials.

File: source/includes/link-user-identities.rst [hunk 2]
```diff
@@ -0,0 +1,25 @@
+Realm provides :ref:`many authentication providers <authentication-providers>` 
+to log users into your app. Each provider creates a unique user identity. 
+Realm lets you merge multiple credentials into one user identity.
+
+However, you can only link credentials that are not already in use by another
```
Comment (mongomoe) [inline source/includes/link-user-identities.rst:5]: Suggestion/Not Required for LGTM: 

I don't like the use of "However" in docs, it feels very "essay-like", can we rephrase this to sound more natural, maybe:

"Credentials must be linked prior to logging a user in, if a user logs in then you cannot link that credential with another credential. Also, you cannot link multiple email/password credentials together."

Comment (cbullinger) [reply source/includes/link-user-identities.rst:5]: Agreed. Updating to avoid 'however'.

**Review by mongomoe — COMMENTED** [review]
1 needed fix + 1 suggestion, close to LGTM. Great work so far

**Review by mongomoe — APPROVED** [review]
LGTM + 1 more suggestion
